config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -101,3 +113,199 @@ const (
+	// The current default value is `kubernetes.io/os: linux`.
+	// +optional
+	// +kubebuilder:validation:MinProperties=1
+	// +kubebuilder:validation:MaxProperties=10
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// resources defines the compute resource requests and limits for the Alertmanager container.
+	// This includes CPU, memory and HugePages constraints to help control scheduling and resource usage.
+	// When not specified, defaults are used by the platform. Requests cannot exceed limits.
+	// This field is optional.
+	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	// This is a simplified API that maps to Kubernetes ResourceRequirements.
+	// The current default values are:
+	//   resources:
+	//     requests:
+	//       cpu: 4m
+	//       memory: 40Mi
+	// +optional
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=10
+	Resources []ContainerResource `json:"resources,omitempty"`
+	// secrets Defines a list of secrets that need to be mounted into the Alertmanager.
+	// The secrets must reside within the same namespace as the Alertmanager object.
+	// They will be added as volumes named secret-<secret-name> and mounted at
+	// /etc/alertmanager/secrets/<secret-name> within the 'alertmanager' container of
+	// the Alertmanager Pods.
+	//
+	// These secrets can be used to authenticate Alertmanager with endpoint receivers.
+	// For example, you can use secrets to:
+	// - Provide certificates for TLS authentication with receivers that require private CA certificates
+	// - Store credentials for Basic HTTP authentication with receivers that require password-based auth
+	// - Store any other authentication credentials needed by your alert receivers
+	//
+	// This field is optional.
+	// Maximum length for this list is 10
+	// +optional
+	// +kubebuilder:validation:MaxItems=10
+	// +listType=atomic
+	Secrets []SecretName `json:"secrets,omitempty"`
+	// tolerations is a list of tolerations applied to network diagnostics components
+	// tolerations is optional.
+	//
+	// When omitted, this means the user has no opinion and the platform is left
+	// to choose reasonable defaults. These defaults are subject to change over time.
+	// Maximum length for this list is 10
+	// +kubebuilder:validation:MaxItems=10
+	// +listType=atomic
+	// +optional
+	Tolerations []v1.Toleration `json:"tolerations,omitempty"`
+	// topologySpreadConstraints defines rules for how Alertmanager Pods should be distributed
+	// across topology domains such as zones, nodes, or other user-defined labels.
+	// topologySpreadConstraints is optional.
+	// This helps improve high availability and resource efficiency by avoiding placing
+	// too many replicas in the same failure domain.
+	//
+	// When omitted, this means no opinion and the platform is left to choose a default, which is subject to change over time.
```

Comment:
What is the default topology spread constraints?

Reply:
The current default is an empty list?